pkg/controller/clusterinst_api.go

```diff
@@ -788,121 +789,124 @@ func (s *ClusterInstApi) createClusterInstInternal(cctx *CallContext, in *edgepr
+					// to support multiple optional resources from
+					// multiple node pools.
+					if in.OptRes != "" && optRes != in.OptRes {
+						return fmt.Errorf("cluster currently only supports a single cluster-wide optional resource, but have both %q and %q", in.OptRes, optRes)
+					}
+					if in.AvailabilityZone != "" && az != in.AvailabilityZone {
+						return fmt.Errorf("availability zone mismatch, flavors selected from both %s and %s zones", in.AvailabilityZone, az)
+					}
+					in.OptRes = optRes
+					in.AvailabilityZone = az
+					log.SpanLog(ctx, log.DebugLevelApi, "Selected Cloudlet Node Flavor", "pool", pool)
 				}
-				if in.AvailabilityZone != "" && az != in.AvailabilityZone {
-					return fmt.Errorf("availability zone mismatch, flavors selected from both %s and %s zones", in.AvailabilityZone, az)
+			} else {
+				az, optRes, err := s.setInfraFlavor(ctx, ostm, &cloudlet, features, &info, in.NodeResources)
+				if err != nil {
+					return fmt.Errorf("failed to select infra flavor, %s", err)
 				}
 				in.OptRes = optRes
 				in.AvailabilityZone = az
-				log.SpanLog(ctx, log.DebugLevelApi, "Selected Cloudlet Node Flavor", "pool", pool)
-			}
-		} else {
-			az, optRes, err := s.setInfraFlavor(ctx, ostm, &cloudlet, features, &info, in.NodeResources)
-			if err != nil {
-				return fmt.Errorf("failed to select infra flavor, %s", err)
+				log.SpanLog(ctx, log.DebugLevelApi, "Selected Cloudlet Node Flavor", "res", in.NodeResources)
 			}
-			in.OptRes = optRes
-			in.AvailabilityZone = az
-			log.SpanLog(ctx, log.DebugLevelApi, "Selected Cloudlet Node Flavor", "res", in.NodeResources)
-		}
-		// Handle control (master) node specification
-		// Note for platforms that manage the control nodes themselves
-		// like Azure AKS, etc, we don't need to configure any master
-		// node resources.
-		if in.Deployment == cloudcommon.DeploymentTypeKubernetes && !features.ManagesK8SControlNodes && !features.NoClusterSupport {
-			// For platforms that do not manage the master nodes,
-			// we need to set the master node flavor so we can create
-			// the master nodes.
-			// Note: temporarily drop support for running workloads on
-			// master nodes. To do so we should allow one of the user
-			// specified node pools to become the master pool.
-			// That requires more work in the platform code to support
-			// such a change and allow for more than 1 master node.
-			var masterNodeResources *edgeproto.NodeResources
-			settings := s.all.settingsApi.Get()
-			if settings.MasterNodeFlavor != "" {
-				masterFlavor := edgeproto.Flavor{}
-				masterFlavorKey := edgeproto.FlavorKey{}
-				masterFlavorKey.Name = settings.MasterNodeFlavor
-				if s.all.flavorApi.store.STMGet(stm, &masterFlavorKey, &masterFlavor) {
-					masterNodeResources = masterFlavor.ToNodeResources()
+			// Handle control (master) node specification
+			// Note for platforms that manage the control nodes themselves
+			// like Azure AKS, etc, we don't need to configure any master
```

Comment:
Oh wow - git got sooo confused here :)

Reply:
Yeah it's not very good at figuring out a whole bunch of code just got indented.